src/components/AddProduct.tsx

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+
+const AddProduct = () => {
+  const [name, setName] = React.useState("");
+  const [price, setPrice] = React.useState("");
+  const [category, setCategory] = React.useState("");
+  const addProduct = async () => {
+    console.log(name, price, category);
+
+    let result = await fetch("http://localhost:3000/product", {
+      method: "post",
+      body: JSON.stringify({ name, price, category }),
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+    result = await result.json();
+    console.log(result);
+  };
+
+  return (
+    <div className="product">
+      <h1>Add Producct</h1>
+      <input
+        type="text"
+        placeholder="Product Name"
+        className="inputbox"
+        value={name}
+        onChange={(e) => {
+          setName(e.target.value);
```

Comment:
unnecessary use of state management for react prefer the form onSubmit event
https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/submit_event

Always go for web api rather than using core react.